client/blocks/editor-checkout-modal/style.scss

```diff
@@ -5,7 +5,22 @@
 .editor-checkout-modal {
 	z-index: z-index( 'root', '.main.calypsoify.is-iframe' );
 
+	.components-modal__header {
+		padding: 0 22px;
+
+		.components-modal__icon-container > svg {
+			vertical-align: middle;
+			fill: var( --color-text );
+		}
+
+		// hide border on desktop
+		@include break-medium {
+			border: none;
+		}
+	}
```

Comment:
Since we're using the same styling in the help center modal, we can create a mixin and use it in both. WDYT?